cli-plugins/plugin/plugin.go

```diff
@@ -27,21 +27,15 @@ import (
 var PersistentPreRunE func(*cobra.Command, []string) error
 
 // RunPlugin executes the specified plugin command
-func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager.Metadata) error {
-	tcmd := newPluginCommand(dockerCli, plugin, meta)
+func RunPlugin(ctx context.Context, dockerCli *command.DockerCli, plugin *cobra.Command, meta manager.Metadata) error {
+	tcmd := newPluginCommand(ctx, dockerCli, plugin, meta)
 
 	var persistentPreRunOnce sync.Once
 	PersistentPreRunE = func(cmd *cobra.Command, _ []string) error {
 		var err error
 		persistentPreRunOnce.Do(func() {
-			cmdContext := cmd.Context()
-			// TODO: revisit and make sure this check makes sense
-			// see: https://github.com/docker/cli/pull/4599#discussion_r1422487271
-			if cmdContext == nil {
-				cmdContext = context.TODO()
-			}
-			ctx, cancel := context.WithCancel(cmdContext)
-			cmd.SetContext(ctx)
+			ctx2, cancel := context.WithCancel(cmd.Context())
```

Comment:
no i think we should use the cobra context here, since cobra context will be set to the context passed in to the `RunPlugin` function here https://github.com/docker/cli/pull/4774/files/7f65f3a4ffc644ac23f4c519ac3d98971ca61ccb#diff-5525052fe59ca68d6a9837832220a8c362270aa47a8520c1e2e4e43798c41dc8R27

Reply:
Maybe another way to think about this is "if not for setting the context that is going to be used for my commands later on, _**what is**_ happening with this `ctx`? What happens if I cancel it?
cc @thaJeztah

Reply:
Oh I misread the change (that's what I get for reviewing from my phone) 😓 

Sounds good, ty!